### DISCUSSION TITLE
Error: spawn docker-compose ENOENT

### DISCUSSION THREAD

**Discussion by qingxuantang** [reworkd/AgentGPT#756] [Q&A] [answered]
Hello all!

I have been trying to deploy with ./setup.sh --docker and this error was raised:

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn docker-compose ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn docker-compose',
  path: 'docker-compose',
  spawnargs: [ 'up', '--build' ]
}

Node.js v18.16.0

Here is my environment:
node version:v18.16.0
Docker version 24.0.2, build cb74dfc
Ubuntu  22.04 64bit
ecs instance on Aliyun

Please help!

**Comment by qingxuantang** [comment, marked as answer]
OK... I resolved this error with the following code:

sudo apt install docker-compose

Then cd into next folder and remove node_modules folder in case there's any corrupted modules; and reinstall:
cd next
sudo rm -r node_modules
npm install

Then update docker-compose and docker.
sudo apt install --upgrade docker-compose
sudo apt install --upgrade docker

The agent is now up and running. If anyone encountered something similar, hope this may help.

**Reply by adunderground** [reply]
This is helpful, thanks!

**Comment by Vodolay** [comment]
Same error.

Node: 19.9.0
Docker: 24.0.2
Docker compose: 2.18.1

.bashrc has the line 
alias docker-compose="docker compose --compatibility $@"

Still, the error persists. And installing Compose v1 is not an appealing option.
